### PR TITLE
README.md: reorganize README

### PR DESCRIPTION
**Pull request by ericchiang**
* Highlights that dex is NOT a user-management system.
* Highlights ID Tokens as dex's primary feature.
* General cleanup.

cc @ElijahCaine @radhikapc @rithujohn191 @squat @philips

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -6,20 +6,45 @@
 
 ![logo](Documentation/logos/dex-horizontal-color.png)
 
-Dex is an OpenID Connect server that connects to other identity providers. Clients use a standards-based OAuth2 flow to login users, while the actual authentication is performed by established user management systems such as Google, GitHub, FreeIPA, etc.
+Dex is an identity service that uses [OpenID Connect][openid-connect] to drive authentication for other apps.
 
-[OpenID Connect][openid-connect] is a flavor of OAuth that builds on top of OAuth2 using the JOSE standards. This allows dex to provide:
+Dex is NOT a user-management system, but acts as a portal to other identity providers through "connectors." This lets dex defer authentication to LDAP servers, SAML providers, or exstablished identity providers like GitHub, Google, and Active Directory. Clients write their authentication logic once to talk to dex, then dex handles the protocols for a given backend.
```
Comment (radhikapc) [inline README.md]: a typo : exstablished

File: README.md [hunk 2]
```diff
@@ -6,20 +6,45 @@
...
 
-* Short-lived, signed tokens with standard fields (such as email) issued on behalf of users.
-* "well-known" discovery of OAuth2 endpoints.
-* OAuth2 mechanisms such as refresh tokens and revocation for long term access.
-* Automatic signing key rotation.
+## ID Tokens
 
-Standards-based token responses allows applications to interact with any OpenID Connect server instead of writing backend specific "access_token" dances. Systems that can already consume ID Tokens issued by dex include:
+ID Tokens are an OAuth2 extension introduced by OpenID Connect and dex's primary feature. ID Tokens are [JSON Web Tokens][jwt-io] (JWTs) signed by dex and returned as part of the OAuth2 response that attest to the end user's identity. An example JWT might look like:
```
Comment (radhikapc) [inline README.md]: JWT - JSON Web Tokens ? please expand at the first occurrence of an abbreviation.

Comment (ericchiang) [inline README.md]: It already is

> [JSON Web Tokens][jwt-io] (JWTs)

File: README.md (hunk 1 above)
Comment (ericchiang) [inline README.md]: thanks! fixed.

**Pull request merged**